torch/csrc/jit/codegen/cuda/scheduler/utils.cpp

```diff
@@ -1644,6 +1640,25 @@ size_t expandVectorizationToContigMergedDomains(
     return default_word_size;
   }
 
+  size_t max_expand_size = SchedulerRuntimeInfo::max_alignment_size_in_byte;
+  size_t common_alignment_size =
+      SchedulerRuntimeInfo::max_alignment_size_in_byte;
+
+  for (auto inp_out : vectorizable_inputs_outputs) {
+    max_expand_size = std::min(
+        max_expand_size,
+        SchedulerRuntimeInfo::max_alignment_size_in_byte /
+            dataTypeSize(inp_out->dtype()));
+    max_expand_size = std::min(
+        max_expand_size, runtime_info.getMaxVectorizableWidth(inp_out));
+    common_alignment_size =
+        std::min(common_alignment_size, runtime_info.getAlignmentSize(inp_out));
+  }
+
+  if (max_expand_size == default_word_size) {
+    return default_word_size;
+  }
```

Comment:
This seems to make sense, but I wonder if we need to get `max_expand_size`. Does bailing out when it's equal to `default_word_size` avoid some failure?

Reply:
It doesn't avoid a failure, just doesn't make sense to do more complex checks if there isn't any opportunity to increase the vector size. It's just an early bailout on what I think is a fast/easy check.

Reply:
Adding comment for this as well.